src/Microsoft.NetFramework.Analyzers/Core/Helpers/SecurityDiagnosticHelpers.cs

```diff
@@ -184,9 +186,9 @@ private static bool IsSpecifiedProperty(IPropertySymbol symbol, INamedTypeSymbol
             return false;
         }
 
-        private static bool IsSpecifiedPropertyDerived(IPropertySymbol symbol, INamedTypeSymbol namedType, string propertyName)
+        private static bool IsSpecifiedPropertyDerived(IPropertySymbol symbol, INamedTypeSymbol? namedType, string propertyName)
```

Comment:
```suggestion
        private static bool IsSpecifiedPropertyDerived([NotNullWhen(returnValue: true)] IPropertySymbol? symbol, [NotNullWhen(returnValue: true)] INamedTypeSymbol? namedType, string propertyName)
```

Reply:
I also see a bunch of methods below which have null checks, but the parameter types are not decorated with `?`. Probably want to address those?